### PR TITLE
Add support for Immutables: `setimmutable` and `loadimmutable`

### PR DESCRIPTION
**Pull request by Glitch18**
This PR adds support for immutables by implementing two new builtins, `SetImmutable` and `LoadImmutable`.
This also adjusts the parsing mechanism a little by changing how strings are handled in case of `setimmutable` and `loadimmutable`. 

This also adds a test for verification of the working of above functions.

### CODE REVIEW

File: src/warp/yul/storage_access.py [hunk 1]
```diff
@@ -28,10 +28,13 @@ def generate_setter_body(setter_var: str, args: Iterable[str]) -> str:
...
-    warp_assert(
-        all(x == "Uint256" for x in var.arg_types),
-        "We don't support storage variables parameterized by types other than Uint256",
-    )
+    # Commenting out the following assert for implementation of immutables,
+    # which uses a felt to Uint256 storage map.
+
+    # warp_assert(
+    #     all(x == "Uint256" for x in var.arg_types),
+    #     "We don't support storage variables parameterized by types other than Uint256",
```
Comment (murcake) [inline src/warp/yul/storage_access.py:36]: You can just remove it completely

File: tests/behaviour/immutables_test.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        [cairo], debug_info=True, cairo_path=[CAIRO_PATH]
+    )
+
+    starknet = await StarknetState.empty()
+    contract_address = await deploy_contract(
+        starknet, program_info, contract_definition, 500
+    )
+
+    res = await invoke_method(starknet, program_info, contract_address, "MY_UINT")
+    assert res.retdata == [32, 2, 0, 500]
```
Comment (murcake) [inline tests/behaviour/immutables_test.py:36]: Let's also check that MY_ADDRESS is zero

**Review by murcake — APPROVED** [review]

File: src/warp/yul/storage_access.py (hunk 1 above)
Comment (Glitch18) [reply src/warp/yul/storage_access.py]: Done 👍

File: tests/behaviour/immutables_test.py (hunk 2 above)
Comment (Glitch18) [reply tests/behaviour/immutables_test.py:36]: Added a check for this 👍

**Comment by murcake** [comment]
Can you please also run `black .`?

File: src/warp/yul/BuiltinHandler.py [hunk 3]
```diff
@@ -730,8 +730,15 @@ class LinkerSymbol(NotImplementedStarkNet):
...
-class LoadImmutable(NotImplementedStarkNet):
-    pass
+class LoadImmutable(DynamicHandler):
+    def __init__(self, cairo_functions: CairoFunctions, imm_dict: dict[str, int]):
+        super().__init__(cairo_functions.load_immutable_function)
+        self.imm_dict = imm_dict
+
+    def get_function_call(self, function_args: Sequence[str]) -> str:
+        (arg,) = function_args
+        id = self.imm_dict.get(arg)
```
Comment (murcake) [inline src/warp/yul/BuiltinHandler.py:740]: This causes a typing error. `imm_dict.get` might return `none` if there's no value associated with `key`. In that case, we should throw an exception to the user

**Comment by murcake** [comment]
https://github.com/NethermindEth/warp/runs/5020117986?check_suite_focus=true

File: src/warp/yul/parse.py [hunk 4]
```diff
@@ -36,11 +36,15 @@ def parse_node(yul_ast) -> ast.Node:
         return parser(yul_ast)
 
 
-def parse_expression(yul_ast) -> ast.Expression:
+def parse_expression(yul_ast, func_name: str = None) -> ast.Expression:
```
Comment (murcake) [inline src/warp/yul/parse.py:39]: `func_name: Optional[str]`

**Pull request merged**